test/ex_doc/autolink_test.exs

```diff
@@ -26,22 +26,35 @@ defmodule ExDoc.AutolinkTest do
                ~m"[`IEx.Helpers`](https://hexdocs.pm/iex/IEx.Helpers.html)"
     end
 
-    test "private module" do
-      assert_unchanged("String.Unicode")
+    test "hidden module" do
+      captured =
+        assert_warn(fn ->
+          assert_unchanged("String.Unicode")
+        end)
+
+      assert captured =~
+               "documentation references module \"String.Unicode\" but it is hidden"
     end
 
     test "erlang module" do
       assert_unchanged(":array")
     end
 
     test "unknown module" do
-      assert_unchanged("Unknown")
+      captured =
+        assert_warn(fn ->
+          assert_unchanged("Unknown")
```

Comment:
if we expect not to warn, let’s remove assert warn. Otherwise, since we “refute =~”, if the warning message ever changes this test won’t catch that.

Reply:
done